src/platform/plugins/shared/dashboard/public/dashboard_api/load_dashboard_api/load_dashboard_api.ts

```diff
@@ -44,25 +39,10 @@ export async function loadDashboardApi({
     return;
   }
 
-  // --------------------------------------------------------------------------------------
-  // Combine saved object state and session storage state
-  // --------------------------------------------------------------------------------------
-  const sessionStorageInput = ((): Partial<DashboardState> | undefined => {
-    if (!creationOptions?.useSessionStorageIntegration) return;
-    return getDashboardBackupService().getState(savedObjectResult.dashboardId);
-  })();
-
-  const combinedSessionState: DashboardState = {
-    ...(savedObjectResult?.dashboardInput ?? {}),
-    ...sessionStorageInput,
-  };
-  combinedSessionState.references = sessionStorageInput?.references?.length
-    ? sessionStorageInput?.references
-    : savedObjectResult?.references;
```

Comment:
Are we overwriting sessionStorage references (unsaved changes) with saved object references?